apm-agent-plugins/apm-profiling-plugin/src/main/java/co/elastic/apm/agent/profiler/CallTree.java

```diff
@@ -391,6 +414,25 @@ public void resetState() {
         }
     }
 
+    public void addChildId(long id) {
+        if (childIds == null) {
+            childIds = new LongList();
```

Comment:
Doesn't worth pooling?

Reply:
Not really. Lists are quite tricky to pool as they can have vastly different sizes. The danger is that over time, all lists in the pool will be quite large even though there are only few that actually contain a lot of elements.
Judging from the benchmarks, the allocations caused by the inferred spans processing increase by 10%. That seems reasonable and it's definitely not the major source of allocations.